### PR TITLE
Update remote names

### LINKED ISSUE #415: Update remote names in docs
Currently, some documentation uses `origin` to refer to the main repo while other documentation uses it to refer to one's fork of the main repo. We can clean this up by using the generic terms `<upstream-origin>` and `<fork-origin>`.

### PR DESCRIPTION
**Pull request by forsyth2**
Update remote names in the docs. Resolves #415.

### CODE REVIEW

File: docs/source/contributing.rst [hunk 1]
```diff
@@ -83,7 +83,7 @@ To begin editing:
 
       Docs version selector dropdown in the bottom left-hand corner
 
-7. Create a pull request from ``your-fork/e3sm_diags/branch-name`` to ``E3SM-Project/e3sm_diags/master``.
+7. Create a pull request from ``<your-fork>/e3sm_diags/branch-name`` to ``E3SM-Project/e3sm_diags/master``.
```
Comment (forsyth2) [inline docs/source/contributing.rst:86]: Brackets should be added to other appearances of `<your-fork>`

**Comment by forsyth2** [comment]
@tomvothecoder This is ready for review. Hopefully I haven't mixed up any `<fork-origin>` and `<upstream-origin>` references. I ran `git grep origin` and `git grep upstream` in `docs/source` and it looks like I updated everything that needed to be updated.

File: docs/source/dev_guide/project-standards.rst [hunk 2]
```diff
@@ -93,17 +93,17 @@ How to squash and rebase commits
...
+    git rebase <upstream-origin>/master
+    git push -f <fork-origin> master
 
 2. Rebase branch onto ``master`` ::
 
-    git checkout branchName
-    git rebase master
-    git push -f origin branchName
+    git checkout <branch-name>
+    git rebase master # If you have merge conflicts, it may be easier to `git rebase --abort` and first do step 4 with SHA of the last commit before your work
```
Comment (forsyth2) [inline docs/source/dev_guide/project-standards.rst:105]: @tomvothecoder I'll just leave this line in here until one of us changes the order of the steps in a future pull request.

Comment (tomvothecoder) [reply docs/source/dev_guide/project-standards.rst:105]: #418 will update this

**Review by tomvothecoder — APPROVED** [review]
Looks good, minor comment. Thanks

File: docs/source/install.rst [hunk 3]
```diff
@@ -164,31 +164,44 @@ Furthermore, the dev environment includes QA tools such as code formatters, lint
...
-        # aren't already in upstream/master are replayed on top of the
-        # other branch:
-        git rebase upstream/master
-        git push -f origin master
+        # aren't already in <upstream-origin>/master are replayed on top of that branch:
+        git rebase <upstream-origin>/master
+
+        # Push your master branch to your GitHub fork:
+        # Note that <remote-origin> should be `origin` if you cloned your fork as above.
+        git push -f <remote-origin> master
```
Comment (tomvothecoder) [inline docs/source/install.rst:204]: Should be <fork-origin> instead of <remote-origin>?

Comment (tomvothecoder) [inline docs/source/install.rst:204]: Should be "<fork-origin>"?

**Pull request merged**